Fix several occasional fatal errors

MrPetovan opened the pull request:
Address:
- https://github.com/friendica/friendica/issues/9250#issuecomment-743382769
- https://github.com/friendica/friendica/issues/9250#issuecomment-743407998
- https://github.com/friendica/friendica/issues/9250#issuecomment-743769813

MrPetovan committed:
Simplify Contact::addRelationship call in ActivityPub\Processor::followUser

- Address https://github.com/friendica/friendica/issues/9250#issuecomment-743382769

MrPetovan committed:
Account for missing contact in dfrn_request_post

- Address https://github.com/friendica/friendica/issues/9250#issuecomment-743407998

MrPetovan committed:
Return early if an URI id cannot be obtained in ActivityPub\Processor::createItem

- Address https://github.com/friendica/friendica/issues/9250#issuecomment-743769813

annando commented on src/Protocol/ActivityPub/Processor.php:
```diff
@@ -903,20 +907,15 @@ public static function followUser($activity)
 		self::switchContact($item['author-id']);
 
-		$result = Contact::addRelationship($owner, $contact, $item, false, $note);
+		$result = Contact::addRelationship($owner, [], $item, false, $activity['content'] ?? '');
```
Why have you removed the contact array from here?

MrPetovan replied:
Because it is also retrieved based on the `author-id` key of the `$item` in `Contact::addRelationship` and overwrites the `$contact` parameter, which made the retrieval in this method completely superfluous whether it failed or succeeded.

annando replied:
The `author-id` is the public contact, that record is the personal contact. Although there is some other fallback in the code, I would like to keep that code in the RC phase since it isn't testable so easily.

MrPetovan replied:
Here's the full rundown of how it was working (or not, sometimes) in this particular instance.

First, in `src/Protocol/ActivityPub/Processor.php`:
1. [Line 892](https://github.com/friendica/friendica/blob/ef4f192a75ee4499f85db3491fae0e0917f11bf2/src/Protocol/ActivityPub/Processor.php#L892): `$uid` is obtained from the activity object ID which is the importer URL.
2. [Line 897](https://github.com/friendica/friendica/blob/ef4f192a75ee4499f85db3491fae0e0917f11bf2/src/Protocol/ActivityPub/Processor.php#L897): `$owner` is obtained from `$uid`. It's the importer local user array.
3. [Line 902](https://github.com/friendica/friendica/blob/ef4f192a75ee4499f85db3491fae0e0917f11bf2/src/Protocol/ActivityPub/Processor.php#L902): `$cid` is obtained from the activity actor URL and `$uid`. It's a contact ID specific to the importer user.
4. [Line 906](https://github.com/friendica/friendica/blob/ef4f192a75ee4499f85db3491fae0e0917f11bf2/src/Protocol/ActivityPub/Processor.php#L906): `$contact` is populated from this same `$cid` so it's the user-specific contact data array.
5. [Line 911](https://github.com/friendica/friendica/blob/ef4f192a75ee4499f85db3491fae0e0917f11bf2/src/Protocol/ActivityPub/Processor.php#L911): `$item` is populated from the public contact ID from the activity actor URL.

Now we go over to `src/Model/Contact.php`:
1. [Line 2386](https://github.com/friendica/friendica/blob/c787281effb66fdfe9fc52ac0ae87dcfe60a10f9/src/Model/Contact.php#L2386): At this point, `$importer` is the local user retrieved earlier, `$contact` has the user-specific contact data if it exists, and `$item` has the public contact ID.
2. [Line 2412](https://github.com/friendica/friendica/blob/c787281effb66fdfe9fc52ac0ae87dcfe60a10f9/src/Model/Contact.php#L2412): `$cid` is populated with the contact ID specific to the importer user.
3. [Line 2414](https://github.com/friendica/friendica/blob/c787281effb66fdfe9fc52ac0ae87dcfe60a10f9/src/Model/Contact.php#L2414): If this contact exists, `$contact` is then overridden with the user-specific contact data, which it already contained if it existed.

The logical conclusion is that the user-specific contact data retrieval in  `src/Protocol/ActivityPub/Processor.php:906` is superfluous because, in this specific case, the data we pass will be overridden with the same data in `src/Model/Contact.php:2414`. Of course I could have added a simple variable type check to `$contact` before passing it to `Contact::addRelationship`, but in this case the `$contact` variable isn't useful at all since it's overridden with the same exact data in the called `Contact` method. Now it's out of the question to change the general `Contact::addRelationship` method, but we can definitely change the way we call it in `Processor::followUser` without any side-effect.

annando replied:
We really should change this function in the future since the parameters are really confusing.

MrPetovan committed:
Account for inbox array in Worker/APDelivery

- Address https://github.com/friendica/friendica/issues/9250#issuecomment-744339671

MrPetovan commented:
Adresses https://github.com/friendica/friendica/issues/9250#issuecomment-744339671 as well

annando commented:
I will have a look at it later this evening.

AlfredSK commented:
Pretty good so far. But not quite yet. I have a new fatal when running the worker now. Reported at the issue for fatal errors...

MrPetovan commented:
Thanks, there seems to be a concerning issue between the parameters passed to the `apdelivery` worker task creation, and the parameters actually expected by the task itself. I believe it has to do with https://github.com/friendica/friendica/pull/9651 that changed the `$inbox` variable type.
